common/util/autoware_debug_tools/src/stop_reason2pose.py

```diff
@@ -0,0 +1,159 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import argparse
+import math
+import sys
+
+import numpy as np
+import rclpy
+from rclpy.node import Node
+from autoware_planning_msgs.msg import StopReasonArray
+from geometry_msgs.msg import PoseStamped
+from rtree import index
+
+from self_pose_listener import SelfPoseListener
+from case_converter import pascal2snake
+
+
+class StopReason2PoseNode(Node):
+    def __init__(self, options):
+        super().__init__('stop_reason2pose_node')
+        self._options = options
+        self._sub_pose = self.create_subscription(
+            StopReasonArray, self._options.topic_name, self._on_stop_reasons, 100)
+        self._pub_pose_map = {}
+        self._idx_map = {}
+        self._pose_map = {}
+        self._self_pose_listener = SelfPoseListener()
+
+    def _on_stop_reasons(self, msg):
+        for stop_reason in msg.stop_reasons:
+            snake_case_stop_reason = pascal2snake(stop_reason.reason)
+
+            for stop_factor in stop_reason.stop_factors:
+                pose = PoseStamped()
+                pose.header = msg.header
+                pose.pose = stop_factor.stop_pose
+
+                # Get nearest pose
+                th_dist = 1.0
+                nearest_pose_id = self._get_nearest_pose_id(
+                    snake_case_stop_reason, pose.pose, th_dist)
+                if nearest_pose_id:
+                    self._update_pose(snake_case_stop_reason, pose.pose, nearest_pose_id)
+                    pose_id = nearest_pose_id
+                else:
+                    pose_id = self._register_pose(snake_case_stop_reason, pose.pose)
+
+                pose_topic_name = "{snake_case_stop_reason}_{pose_id}".format(**locals())
+                if pose_topic_name not in self._pub_pose_map:
+                    self._pub_pose_map[pose_topic_name] = self.create_publisher(
+                        PoseStamped, "~pose" + pose_topic_name, queue_size=1)
+                self._pub_pose_map[pose_topic_name].publish(pose)
+
+            # Publish nearest stop_reason without number
+            self_pose = self._self_pose_listener.get_current_pose()
+
+            nearest_pose = PoseStamped()
+            nearest_pose.header = msg.header
+            nearest_pose.pose = self._get_nearest_pose_in_array(stop_reason, self_pose.pose)
+
+            if nearest_pose.pose:
+                if snake_case_stop_reason not in self._pub_pose_map:
+                    self._pub_pose_map[snake_case_stop_reason] = self.create_publisher(
```

Comment:
create_publisher has not "queue_size" arugument

Reply:
fixed!